python-package/setup.py

```diff
@@ -28,6 +28,7 @@
       install_requires=[
           'numpy',
           'scipy',
+          'scikit-learn',
```

Comment:
Note that it is .sklearn instead of sklearn. Sklearn dependency has been optional for a while.